Change check fail class and print method (#114)
hubverse-org/hubValidations: vignettes/articles/hub-validations-class.Rmd

- If a check is skipped, a `<message/check_info>` condition class object is returned.

- Checks vary with respect to whether they return an `<error/check_error>` or `<error/check_failure>` condition class object if the check fails. 
  - **`<error/check_error>` ** class objects indicate early termination of the validation process because of failure of a check downstream checks depend on.

>   - **`<error/check_failure>`** class objects indicate a check that failed but the failure does not affect downstream checks so validation was able to proceed.

Author: I change the ordering but kept the original comments in a6058b99dfc1122ca26b40d73c12f0d555ff2f5f

> Ultimately, both will cause overall validation to fail and the two classes are used primarily to communicate early termination.

Reviewer: ```suggestion
Ultimately, both will cause overall validation to fail. The `<error/check_error>` class exists to alert you to the fact that _there may be more errors not yet reported_ due to early termination of the check process.
```

## `hub_validations` print method

`hub_validations` objects have their own print method which displays the result, the file name and message of each check:

- `✔` indicates a check was successful (a `<message/check_success>` condition class object was returned)

> - `ⓧ` indicates a check that downstream checks depend on failed causing early return. (a `<error/check_error>` condition class object was returned)
> - `✖` indicates a check failed but does not affect downstream checks so validation was able to proceed (a `<error/check_failure>` condition class object was returned)

Reviewer: ```suggestion
- `✖` indicates a check with no downstream checks failed. (a `<error/check_failure>` condition class object was returned)
- `ⓧ` indicates a check failed and preventing downstream checks from being performed, causing early return. (a `<error/check_error>` condition class object was returned)
```

Author: Reordered and rephrased here: be60ccb4806082f3ebbdbed3eb875eae7ef16183
